### DISCUSSION TITLE
query information from API shopify for data page

### DISCUSSION THREAD

**Discussion by knupel** [gatsbyjs/gatsby#33394] [Help]
Hello, I try to access to data page from `GraphQL`, but I don't know how do that. When I see the list of elements from `Shopify` `GraphQL`, I've a feeling is just possible to access to `collections` and `products`. I hope I'm wrong.
The only hope I had it was `allShopifyMetafieldInterface` but I don't find something interesting in this part.

In my case I want access to the data from the page `café 366` `Professionnels / Entreprises` and `On parle de nous`. So if there is any solution to catch it or a nice guide how explore all data from `GraphQL`that's help me. 

I hope my request is clear.

![shopify_pages](https://user-images.githubusercontent.com/3332269/135634303-2e241483-fe04-468b-874a-f213bd4cdf45.jpg)
![graphql_gatsby](https://user-images.githubusercontent.com/3332269/135634319-0794a9f9-a9a4-45f4-943b-b3203ba55bf7.jpg)

**Comment by knupel** [comment]
I'm back after few months with a same question... But no body answer so maybe is not possible ? Because if it's possible to carry the shopify page not only the product that's can be very nice.

**Reply by unimprobable** [reply]
Hello, the gatsby-source-shopify plugin doesn't have support for pages (or blogs).

You can, however, source these from Shopify's GraphQL api in your gatsby-node.js file or create your own local plugin that does it.

**Reply by knupel** [reply]
Thanks for your answer @unimprobable , it's a pity the plugin don't have this option. That's can be awesome if this feature appear in the future. Unfortunately, I'm not expert with `Javascript`, `React` or `Gatsby` to create a plugin to catch the from `Shopify API`.

In case I try to extract any information from `GraphQL` api in my `gatsby-node.js` by which way I must start ?

**Reply by unimprobable** [reply]
Here's a good start for the basics on how to create a source plugin:

https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/creating-a-source-plugin/

Is there any reason why you need that html from the online store pages over any other source of content? Is it for ease of editing?

**Reply by knupel** [reply]
It's for the customer / user Shopify CMS. If it's possible to catch all information from Shopify: product and pages [see the first screenshot I've added to this post]. That's easier for the final customer to set the content of the site. Because if it's not possible I must add something like Contentful to set the pages. After I must learn to the client to use `Contentful` and `Shopify`so if I can avoid that, that's can be great !  I'm going to look your link to understand the plugin way !

**Reply by knupel** [reply]
In case there is project to add the possibility to grab page with `gatsby-source-shopify` there is this link.
https://shopify.dev/api/admin-rest/2022-01/resources/page#top
